lms/views/api/blackboard/sync.py

```diff
@@ -52,12 +61,27 @@ def _get_blackboard_groups(self):
             if not learner_groups:
                 raise CanvasStudentNotInGroup(group_set=group_set_id)
 
-            return learner_groups
+            groups = self._to_groups_groupings(learner_groups)
+            self._grouping_service.upsert_grouping_memberships(
+                self._user_service.store_lti_user(lti_user), groups
+            )
+            return groups
 
         if grading_student_id := self._request.json.get("gradingStudentId"):
-            return self._blackboard_api.user_groups(
-                course_id, grading_student_id, group_set_id
-            )
+            db = self._grouping_service._db
+
+            authority_provided_id = hashed_id(tool_guid, course_id)
+            course = self._course_service.get(authority_provided_id)
+            return [
+                gm.grouping
+                for gm in db.query(GroupingMembership)
+                .join(User)
+                .filter(
+                    GroupingMembership.grouping_id.in_([g.id for g in course.children]),
```

Comment:
Not sure what would be the right approach but we also want to preload those children in the real version. Maybe a `with_children` bool on `.get()`?